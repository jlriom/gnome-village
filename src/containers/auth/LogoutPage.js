import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import * as authActions from "../../store/actions/authActions";

import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

const LogoutPage = ({ show, handleClose, logout }) => {
	const handleLogout = () => {
		logout();
	};
	return (
		<Modal show={show} onHide={handleClose}>
			<Modal.Header closeButton>
				<Modal.Title>Logout</Modal.Title>
			</Modal.Header>

			<Modal.Body>
				<p>You are going to logout from the application.</p>
				<p>Do you want to continue?</p>
			</Modal.Body>

			<Modal.Footer>
				<Button variant='secondary' onClick={handleClose}>
					Cancel
				</Button>
				<Button variant='primary' onClick={handleLogout}>
					Logout
				</Button>
			</Modal.Footer>
		</Modal>
	);
};

LogoutPage.propTypes = {
	show: PropTypes.bool.isRequired,
	handleClose: PropTypes.func.isRequired,
	logout: PropTypes.func.isRequired
};

export default connect(
	() => ({}),
	{ ...authActions }
)(LogoutPage);
