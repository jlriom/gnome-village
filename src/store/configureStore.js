import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
