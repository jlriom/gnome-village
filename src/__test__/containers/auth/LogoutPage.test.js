import React from "react";
import LogoutPage from "../../../containers/auth/LogoutPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";
const store = configureStore();

const props = {
    show: false,
    handleClose: jest.fn(),
    logout: jest.fn()
}

describe("<LogoutPage />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <LogoutPage {...props} />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
