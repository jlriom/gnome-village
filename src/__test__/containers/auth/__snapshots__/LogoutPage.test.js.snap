// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutPage /> renders ok 1`] = `null`;
